web/entities/sapsystem.go

```diff
@@ -0,0 +1,19 @@
+package entities
+
+import (
+	"time"
+)
+
+type SAPSystemInstance struct {
```

Comment:
I thinks we would need to have the `SAPSystem` and `SAPInstance` structs combo.
Why?
Because we can have some SAP instance with a different SID than the SAP system itself (the database doesn't need to have the same SID than the SAP system).

Later on, 1 SAP system is composed by multiple instances, that can have different SID and type. And even different parameters depending on the type. For example, as `application` type instances don't have any System replication data.

Maybe some kind of polymorfism would be a good idea here.

Reply:
@arbulu89 the idea is that this is how we store the data from the events, the `SAPSystem`, aggregation would come in the service when we will transform (group) instances to a `model.SAPSystem` which will hold many instances.
The database aggregation is quite easy since we will search for a `entities.Host` having the same IP as the `DBHost` field here.
As for the SystemReplication data or other fields the idea was to keep them empty in case of the `application` type and just ignore them when casting that to a model.
Basically, polymorphism will be handled at service level not at persistence level.

Reply:
@fabriziosestito said that way better than me :smile: 

I agree that we need to tell the difference between different models, but this shall be done at service level, this PR only introduces a basic projector so we can move forward with other works

Reply:
@fabriziosestito Ok, fair enough. I don't think it will scale well in the future, because we will need to have SAP system specific data, but I guess we will be able to change it